src/core/math.d

```diff
@@ -125,47 +138,37 @@ extern (C) real rndtonl(real x);
 
 version (LDC)
 {
-    version (MinGW)
+    static if (isRealX87)
     {
+        pragma(inline, true)
         real ldexp(real n, int exp) @safe pure nothrow
         {
-            // The MinGW runtime only provides a double precision ldexp, and
-            // it doesn't seem to reliably possible to express the fscale
-            // semantics (two FP stack inputs/returns) in an inline asm
-            // expression clobber list.
-            version (D_InlineAsm_X86_64)
+            real r = void;
+            asm @safe pure nothrow @nogc
             {
-                asm @trusted pure nothrow
-                {
-                    naked;
-                    push RCX;                // push exp (8 bytes), passed in ECX
-                    fild int ptr [RSP];      // push exp onto FPU stack
-                    pop RCX;                 // return stack to initial state
-                    fld real ptr [RDX];      // push n   onto FPU stack, passed in [RDX]
-                    fscale;                  // ST(0) = ST(0) * 2^ST(1)
-                    fstp ST(1);              // pop stack maintaining top value => function return value
-                    ret;                     // no arguments passed via stack
-                }
-            }
-            else
-            {
-                asm @trusted pure nothrow
-                {
-                    naked;
-                    push EAX;
-                    fild int ptr [ESP];
-                    fld real ptr [ESP+8];
-                    fscale;
-                    fstp ST(1);
-                    pop EAX;
-                    ret 12;
-                }
+                `fildl  %1       # push exp
+                 fxch   %%st(1)  # swap ST(0) and ST(1)
+                 fscale          # ST(0) := ST(0) * (2 ^^ ST(1))
+                 fstp   %%st(1)  # pop and keep ST(0) value on top`
+                : "=st" (r)
+                : "m" (exp), "st" (n)
+                : "flags"; // might clobber x87 flags
```

Comment:
Now used for all targets with x87 reals instead of C `ldexpl` (this also affects Phobos' `std.math.ldexp` as that simly forwards). I haven't benchmarked it but I don't think this can be any faster.

Reply:
Oh and the little `"m" (exp)` input is of course the reason for https://github.com/ldc-developers/ldc/pull/3438. ;)